### ISSUE TITLE
Fatal login result: BadRSA

### ISSUE THREAD

**Issue by neckcen** [dyc3/steamguard-cli#104]
Hello,

Using the .deb provided as release 0.4.0 on debian bullseye.

When trying to link an account the following error is triggered:

```
ERROR - rsa error: reqwest::Error { kind: Decode, source: Error("expected value", line: 1, column: 1) }
ERROR - Fatal login result: BadRSA
```

The full output with debug and stacktrace:

```
Username: foo
Password: 
DEBUG - password is present
ERROR - Fatal login result: OtherFailure(expected value at line 1 column 1)
thread 'main' panicked at 'Failed to log in. Account has not been linked.: OtherFailure(expected value at line 1 column 1)', src/main.rs:212:38
stack backtrace:
   0:     0x55a3925a6dd5 - std::backtrace_rs::backtrace::libunwind::trace::h34055254b57d8e79
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55a3925a6dd5 - std::backtrace_rs::backtrace::trace_unsynchronized::h8f1e3fbd9afff6ec
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55a3925a6dd5 - std::sys_common::backtrace::_print_fmt::h3a99a796b770c360
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55a3925a6dd5 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h32d1f94a80615d18
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55a3923a1c2c - core::fmt::write::h306731c068f7162c
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/fmt/mod.rs:1110:17
   5:     0x55a3925a55d4 - std::io::Write::write_fmt::hd2fa90334eee2a21
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/io/mod.rs:1588:15
   6:     0x55a3925a5fcb - std::sys_common::backtrace::_print::h5abaa2601a852287
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55a3925a5fcb - std::sys_common::backtrace::print::h8d81445442bb638f
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55a3925a5fcb - std::panicking::default_hook::{{closure}}::hcfe804496a9fa747
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:208:50
   9:     0x55a3925a51d9 - std::panicking::default_hook::hbea8e3ccf2ba8901
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:225:9
  10:     0x55a3925a51d9 - std::panicking::rust_panic_with_hook::h7ee9e1a2d0f8975a
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:622:17
  11:     0x55a3925bf068 - std::panicking::begin_panic_handler::{{closure}}::h8ab3b4491718b2c7
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:519:13
  12:     0x55a3925befdc - std::sys_common::backtrace::__rust_end_short_backtrace::hd489062ffa586a9f
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x55a3925bef8d - rust_begin_unwind
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:515:5
  14:     0x55a39231a4b0 - core::panicking::panic_fmt::hca6330e3e14086b4
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/panicking.rs:92:14
  15:     0x55a39231abd2 - core::result::unwrap_failed::h6160d2b4bfbb0e87
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/result.rs:1355:5
  16:     0x55a392627530 - steamguard_cli::main::h79145425dffae1de
  17:     0x55a39260b213 - std::sys_common::backtrace::__rust_begin_short_backtrace::hf53feeb4d8b7f96a
  18:     0x55a392612e61 - main
  19:     0x7f1a4201cd0a - __libc_start_main
  20:     0x55a392329dae - _start
  21:                0x0 - <unknown>
```

**Comment by dyc3** [comment]
I'm not able to reproduce this. It looks like you've posted 2 different failures though.

What do you get for this command? 

```
curl -v -d username=YOUR_STEAM_USERNAME "https://steamcommunity.com/login/getrsakey"
```

It should look something like this:

```json
{
    "success":true,
    "publickey_mod":"9318d3e2f96f824f69340c708e3aa24c4b4ebfc71efca33082967abe92e690c3dfb085623757cb6a50cacbcd8f78996ec30a9906bfb34cec9a7607869dcf4ced243c962de945d15d6e01851146608dc2273839bab75cf97aee0a42a0285d8c757017b81276a95f8f9eef6330fcccdbf93f66b14580abdd0444a6d3fd8f1a7f80d6baeb9d77327a8ab0cc2e6c715d9834a41dca87935c73d37c78edc7ac080c45f0254fb5b16bde4b778559cf5966b30066023cbaad0b60a326b547c6b681aa9fd6bcb7851dc91da3c2f5647ff55fd7225bcd548e39f360df110bb710b31bf87efcc1c6e08dd246173d3316ace47cff56fbf73adaddc689ae162142c209a886bb",
    "publickey_exp":"010001",
    "timestamp":"345744100000",
    "token_gid":"27a..."
}
```

**Comment by neckcen** [comment]
Thank you for looking into it.

>  It looks like you've posted 2 different failures though.

Somehow the error message was slightly different when exporting `RUST_BACKTRACE=full` or not.

Regardless, I believe this might have been fixed by either bb6c733999e197bfa9e537f410fcf42d7eb97c27 or 6d2b10022e19b76c49dff479be47026c99f08d28 because now the output correctly states `DEBUG - saving manifest` and the error is gone.

Thanks!
